packages/core/src/external/aws/location.ts

```diff
@@ -27,20 +33,25 @@ export function getLocationResultPayload({
   return result.Results;
 }
 
-export function getCoordinatesFromLocation({
-  result,
-}: {
-  result: PromiseResult<AWS.Location.SearchPlaceIndexForTextResponse, AWS.AWSError>;
-}): Coordinates {
-  const resp = getLocationResultPayload({ result });
-  const topSuggestion = resp ? resp[0] : undefined;
-  if (topSuggestion) {
-    const point = topSuggestion.Place?.Geometry?.Point;
+export function parseSuggestedAddress(
+  suggestedAddress: AWS.Location.SearchForTextResult
+): GeocodingResult {
+  const point = suggestedAddress.Place?.Geometry?.Point;
+  const relevance = suggestedAddress.Relevance;
+  const label = suggestedAddress.Place.Label;
 
-    if (point) {
-      const lon = point[0];
-      const lat = point[1];
-      if (lon && lat) return { lon, lat };
+  if (point) {
+    const lon = point[0];
+    const lat = point[1];
+    if (lon && lat && relevance && label) {
```

Comment:
nit: could avoid nested 'if' by returning if point is not present before the second check on line 46